src/common/transformations/src/transformations/common_optimizations/reverse_shape_and_type_infer.cpp

```diff
@@ -282,6 +282,15 @@ bool ov::pass::ReverseShapeAndTypeInfer::run_on_model(const std::shared_ptr<ov::
                 if_op->get_input_tensor(0).m_element_type = element::boolean;
                 is_changed = true;
             }
+
+            // in case TensorFlow models, we can deduce predicate shape that must be a scalar
+            // If operations created by fusing Switch-Merge sub-graph contain tf_switch_merge_if rt-info
+            if (if_op->get_rt_info().count("tf_switch_merge_if") &&
```

Comment:
why can't we do it on the FE side?
e.g. inside SwitchMergeResolver?

Reply:
That is because we can't change tensor shape on FE side. ReverseShapeTypeInfer is friendly class for tensor so it is able to change tensor shape. It is more logical to do here because we can deduce shapes for tensors above.